pkg/ext/stores/tokens/tokens.go

```diff
@@ -0,0 +1,1459 @@
+		}
+	}
+
+	return localOptions, nil
+}
+
+// secretFromToken converts the token argument into the equivalent secrets to
+// store in k8s.
+func secretFromToken(token *ext.Token, oldBackendLabels, oldBackendAnnotations map[string]string) (*corev1.Secret, error) {
+	// inject default on creation
+	ttl := token.Spec.TTL
+	if ttl == 0 {
+		ttl = ThirtyDays
+		// pass back to caller (Create)
+		token.Spec.TTL = ttl
+	}
+
+	// annotations -- encode and store into a data field. on reading decode
+	// that field.  this fully separates the annotations on the tokens from
+	// the annotations on the backing secrets.
+
+	annotationBytes, err := json.Marshal(token.Annotations)
+	if err != nil {
+		return nil, err
+	}
+
+	// labels -- encode and store into a data field. on reading decode that
+	// field.  this fully separates the user-visible labels on the tokens
+	// from the labels on the backing secrets.
+
+	labelBytes, err := json.Marshal(token.Labels)
+	if err != nil {
+		return nil, err
+	}
+
+	// labels again -- for proper filtering and searching the referenced
+	// user id, and the kind of the token are placed into labels of the
+	// backing secret -- the keys for these labels are part of the public
+	// API. This may have to be merged with labels set on the secrets by
+	// other apps with access to the secrets.
+
+	backendLabels := oldBackendLabels
+	if backendLabels == nil {
+		backendLabels = map[string]string{}
+	}
+	backendLabels[UserIDLabel] = token.Spec.UserID
+	backendLabels[KindLabel] = token.Spec.Kind
+
+	// ensure that only one of name or generateName is passed through.
+	name := token.Name
+	genName := token.GenerateName
+	if genName != "" {
+		name = ""
+	}
+
+	secret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace:    TokenNamespace,
+			Name:         name,
+			GenerateName: genName,
+			Labels:       backendLabels,
+			Annotations:  oldBackendAnnotations,
+		},
+		StringData: make(map[string]string),
+		Data:       make(map[string][]byte),
+	}
+
+	// system
+	secret.StringData[FieldUID] = string(token.ObjectMeta.UID)
+	secret.StringData[FieldLabels] = string(labelBytes)
+	secret.StringData[FieldAnnotations] = string(annotationBytes)
+
+	// spec
+	enabled := token.Spec.Enabled == nil || *token.Spec.Enabled
+
+	secret.StringData[FieldUserID] = token.Spec.UserID
+	secret.StringData[FieldTTL] = fmt.Sprintf("%d", ttl)
+	secret.StringData[FieldEnabled] = fmt.Sprintf("%t", enabled)
+	secret.StringData[FieldDescription] = token.Spec.Description
+	secret.StringData[FieldKind] = token.Spec.Kind
+	secret.StringData[FieldPrincipalID] = token.Spec.PrincipalID
+
+	lastUsedAsString := ""
+	if token.Status.LastUsedAt != nil {
+		lastUsedAsString = token.Status.LastUsedAt.Format(time.RFC3339)
+	}
+
+	// status
+	secret.StringData[FieldHash] = token.Status.TokenHash
+	secret.StringData[FieldLastUpdateTime] = token.Status.LastUpdateTime
+	secret.StringData[FieldLastUsedAt] = lastUsedAsString
+
```

Comment:
Placeholder for integration of PR #48778.
```
secret.StringData[FieldIdleTimeout] = fmt.Sprintf("%d", token.Status.IdleTimeout)
```
Encode and save new field. Change code as required by the actual field type.